VAS/Areas/VAS/Controllers/CommonController.cs

```diff
@@ -2663,6 +2664,140 @@ private void GenerateShipments_complete(Ctx ctx, ProcessInfo pi, string whereCla
+        /// <param name="qty">Quantity to be matched</param>
+        /// <param name="C_InvoiceLine_ID">C_InvoiceLine_ID</param>
+        /// <param name="trx">Transaction</param>
+        /// <returns>Object of Transaction</returns>
+        /// <author>VIS_427</author>
+        public MTransaction CreateOrUpdateProductTransaction(Ctx ctx, MTransaction mtrx, MInOutLine sLine, bool IsToCreateNew, decimal qty, int C_InvoiceLine_ID,
+            Trx trx)
+        {
+            if (!IsToCreateNew)
+            {
+                mtrx = new MTransaction(ctx, Util.GetValueOfInt(DB.ExecuteScalar("SELECT M_Transaction_ID FROM M_Transaction WHERE" +
+                             " M_InoutLine_ID =" + sLine.GetM_InOutLine_ID())), trx);
+                //set the current quantity after subtracting it from quantity which is to matched
+                mtrx.SetCurrentQty(mtrx.GetCurrentQty() - qty);
+                mtrx.SetMovementQty(sLine.GetMovementQty());
+                //if matching is with invoice then set the invoiceline id
+                if (C_InvoiceLine_ID != 0)
+                {
+                    mtrx.Set_Value("C_InvoiceLine_ID", C_InvoiceLine_ID);
+                }
+            }
+            else
+            {
+                mtrx = new MTransaction(ctx, mtrx.GetAD_Org_ID(), mtrx.GetMovementType()
+                                        , mtrx.GetM_Locator_ID(), mtrx.GetM_Product_ID(), mtrx.GetM_AttributeSetInstance_ID(),
+                                         sLine.GetMovementQty(), mtrx.GetMovementDate(), trx);
+                //set the current quantity after adding it from quantity which is to matched
+                mtrx.SetCurrentQty(sLine.GetMovementQty() + qty);
+                mtrx.SetM_InOutLine_ID(sLine.GetM_InOutLine_ID());
+            }
+            return mtrx;
+        }
+        /// <summary>
+        /// This Function is used to create or update policy tab
+        /// </summary>
+        /// <param name="ctx">Context</param>
+        /// <param name="sLine">Shipment Line Object</param>
+        /// <param name="PolicyDate">PolicyDate</param>
+        /// <param name="toInOutLineMA">Policy Object</param>
+        /// <param name="mtrx">Transaction Object</param>
+        /// <param name="isToCreateNew"></param>
+        /// <param name="trx">Trx</param>
+        /// <returns>Object Of MInoutLineMa</returns>
+        /// <author>VIS_427</author>
+        public MInOutLineMA CreateOrUpdatePolicy(Ctx ctx, MInOutLine sLine, DateTime? PolicyDate, MInOutLineMA toInOutLineMA, MTransaction mtrx, bool isToCreateNew, Trx trx)
+        {
+            if (!isToCreateNew)
+            {
+                int count = DB.ExecuteQuery("UPDATE M_InOutLineMA SET MovementQty=" + sLine.GetMovementQty()
+                    + " WHERE M_InoutLine_ID=" + sLine.GetM_InOutLine_ID(), null, trx);
+                if (count < 0)
+                {
+
```

Comment:
what if the count is less than 0, no logs, atleast write some logs, so that someone can trace later on something